File: lib/Karve/CustomControls/DataGrid/Clases/DataGridColumn/DataGridColumns.vb
Imports Telerik.WinControls.UI

Public Class DataGridColumns
    Inherits System.Collections.Hashtable

    Public Overloads Sub Add(ByVal Columna As Object)
        Try
            Columna.Item = Me.Count
            MyBase.Add(Columna.Name, Columna)
        Catch
        End Try
    End Sub

    Public Function ToArray() As ArrayList
        Dim HC = (From c As Object In Me.Values Order By c.Item Select c)
        Dim RS As New ArrayList
        For Each ctr As Object In HC
            RS.Add(ctr)
        Next
        Return RS
    End Function

End Class

File: src/Controls/KarveControls/DataGrid/DataGridHelpers/DataGridColumns.cs
using System.Collections;

namespace KarveControls.DataGrid.DataGridHelpers
{
    public class DataGridColumns : System.Collections.Hashtable
    {

        public void Add(DataGridColumn Columna)
        {
            try
            {
                Columna.Item = this.Count;
                base.Add(Columna.Name, Columna);
            }
            catch
            {
            }
        }

        public ArrayList ToArray()
        {

            dynamic HC = base.Keys;
            ArrayList RS = new ArrayList();
            foreach (object ctr in HC)
            {
                RS.Add(ctr);
            }
            return RS;
        }

    }
}
